04-onion-routing.md

```diff
@@ -238,11 +236,47 @@ Field descriptions:
      leaking its position in the route.
 
    * `padding`: This field is for future use and also for ensuring that future non-0-`realm`
-     `per_hop`s won't change the overall `hops_data` size.
+     `hop_data`s won't change the overall `hop_payloads` size.
+
+When forwarding HTLCs, nodes MUST construct the outgoing HTLC as specified
+within `hop_data` above; otherwise, deviation from the specified HTLC
+parameters may lead to extraneous routing failure.
+
+### `tlv_payload` payload format
+
+This is a more flexible format, which avoids the redundant `short_channel_id` field for the final node. 
+
+1. tlv: `tlv_payload`
+2. types:
+    1. type: 0 (`destination_signal`)
+    2. data:
+	    * Empty
+    1. type: 2 (`amt_to_forward`)
+    2. data:
+        * [`integer`:`amt_to_forward`]
+    1. type: 4 (`outgoing_cltv_value`)
+    2. data:
+        * [`integer`:`outgoing_cltv_value`]
+    1. type: 6 (`short_channel_id`)
```

Comment:
thoughts?

Reply:
I think this makes sense for TLVs containing a single field, but it might be confusing because TLVs containing multiple fields can't use the same format...

Reply:
was thinking structs could be defined elsewhere, and keep all definitions as a single line, e.g.
```
1. type 42: (`struct`: `struct_name`)
```

Reply:
That's a good idea, a dedicated section that contains all the TLVs used by the onion namespace could be useful.

Reply:
We can change it later, but whoever changes it also has to change the tooling to match.  As long as they do that, I'm happy with any readable format!

But until then, no, it must be parsable by tools/extract-formats.py.